app/models/stage.rb

```diff
@@ -190,4 +192,12 @@ def object_attrs_for_paper_trail(attributes)
   def destroy_deploy_groups_stages
     DeployGroupsStage.where(stage_id: id).delete_all
   end
+
+  def destroy_stage_pipeline
+    project.stages.each do |s|
+      if s.next_stage_ids.delete(id.to_s)
+        s.save!
```

Comment:
if the project has an invalid stage then this would fail ... maybe better to do `save validate: false`  ... also avoid (unlikely) loops with `{project.stages - [self]).each`